aws/logs_monitoring/settings.py

```diff
@@ -189,12 +190,36 @@ def __init__(self, name, pattern, placeholder):
     connect_timeout=5, read_timeout=5, retries={"max_attempts": 2}
 )
 # DD API Key
+# Check if the DD_API_KEY_SECRET_ARN environment variable is set
 if "DD_API_KEY_SECRET_ARN" in os.environ:
     SECRET_ARN = os.environ["DD_API_KEY_SECRET_ARN"]
     logger.debug(f"Fetching the Datadog API key from SecretsManager: {SECRET_ARN}")
-    DD_API_KEY = boto3.client("secretsmanager", config=boto3_config).get_secret_value(
-        SecretId=SECRET_ARN
-    )["SecretString"]
+
+    # Fetch the secret from Secrets Manager
+    secret_response = boto3.client(
+        "secretsmanager", config=boto3_config
+    ).get_secret_value(SecretId=SECRET_ARN)
+
+    # The secret could be either a plain string or a JSON object
+    secret_string = secret_response.get("SecretString")
+
+    try:
+        # Try to parse the secret as JSON
+        secret_json = json.loads(secret_string)
+
+        # If it's a JSON object, look for the 'DD_API_KEY' field
+        if "DD_API_KEY" in secret_json:
+            DD_API_KEY = secret_json["DD_API_KEY"]
+            logger.debug(
+                "Successfully retrieved the Datadog API key from 'DD_API_KEY'."
+            )
+        else:
+            raise ValueError("The secret does not contain the 'DD_API_KEY' field.")
```

Comment:
Just noticed raising a valueerror here is not  properly handled down stream and will lead to an uncaught exception.
I suggest to handle it or simply log an error as we'd still have the `DD_API_KEY` set to the default value `<YOUR_DATADOG_API_KEY>` which won't work anw

Reply:
Updated, lmk if that looks good